types/src/indexer/indexer_db_reader.rs

```diff
@@ -58,4 +58,16 @@ pub trait IndexerReader: Send + Sync {
     ) -> Result<Box<dyn Iterator<Item = Result<(StateKey, StateValue)>> + '_>>;
 
     fn get_latest_internal_indexer_ledger_version(&self) -> Result<Option<Version>>;
+
+    #[cfg(any(test, feature = "fuzzing"))]
+    fn wait_for_internal_indexer(&self, version: Version) -> Result<()> {
```

Comment:
We call this from an async site, should this be async and then we use tokio sleep?

Reply:
I originally wrote it in the way you describe, but felt a little bit uncomfortable to make a trait depending on a specific runtime implementation.